concepts/basics/about.md

```diff
@@ -1,3 +1,230 @@
+    [`range`][man-range] is a function that outputs a stream of numbers
+
+    ```sh
+    $ jq -n 'range(10; 70; 15)'
+    10
+    25
+    40
+    55
+    ```
+
+    Collect them in an array
+    
+    ```sh
+    $ jq -c -n '[range(10; 70; 15)]'
+    [10,25,40,55]
+    ```
+
+~~~~exercism/note
+The comma is not just syntax that separates array elements.
+Comma is an **operator** that joins streams.
+
+For example `[1, 2, 3]` is a filter that uses the array constructor `[]`
+to collect the result of the concatenation of the three expressions `1`, `2`
+and `3`.
+~~~~
+
+### Objects
+
+Similar to many programming languages, use dots to access object properties
+
+```sh
+$ echo '{"foo": {"bar": "qux"}}' | jq '.foo.bar'
+"qux"
+```
+
+Brackets can be used for objects too, but then quotes are needed for string literals:
+
+```sh
+$ echo '{"foo": {"bar": "qux"}}' | jq '.["foo"]["bar"]'
+"qux"
+```
+
+Construct an object with `{}` and `key: value` pairs.
+Quotes are not required around keys that are "simple" strings.
+
+```sh
+jq -n '{question: 6 * 9, answer: 42}'
+```
+outputs
+```none
+{
+  "question": 54,
+  "answer": 42
+}
+```
+
+To treat the key as an expression, you must wrap it in parentheses
+```sh
+echo '[["question", "answer"], [54, 42]]' \
+| jq '{(.[0][0]): .[1][0], (.[0][1]): .[1][1]}'
+```
+
+~~~~exercism/note
+It is quite common to want to extract a subset of keys from a large object:
+for example to extract the `id` and `name` from
+
+```json
+{
+    "id": 101,
+    "name": "alpha widget",
+    "specifications": {...}
+}
+```
+
+We can write
+```jq
+{id: .id, name: .name}
+```
+This is so common, there is shorthand syntax for it:
+```jq
+{id, name}
+```
+~~~~
+
+### Pipelines
+
+For example, given "file.json" containing
+
+```json
+{
+    "key1": "value1",
+    "key2": [5, 15, 25]
+}
+```
+
+Let's now calculate the length of the key2 array:
+
+```sh
+$ echo '{"key1": "value1", "key2": [5, 15, 25]}' | \
+  jq -c '.key2 | length'
+3
+```
+
+We're _piping_ the output of the `.key2` expression as the input to [`length`][man-length], which unsurprisingly outputs the  number of elements in the array.
+
+~~~~exercism/caution
+This is an aspect of `jq` that takes some getting used to --
+most (but not all) functions act like _filters_ where you pass data to the filter's input, not as an argument.
+~~~~
+
+### Filters can ignore their input
+
+For example:
+
+```sh
+jq '6 * 9' <<< '{"answer": 42}'
+```
+
+### Filters can output streams of data
+
+A filter can output more than one value.
+For example, the `.[]` filter outputs each element of an array as a separate value:
+
+```sh
+$ jq -n -c '[1, 2, 3]'
+[1,2,3]
+
+$ jq -n -c '[1, 2, 3] | .[]'
+1
+2
+3
+```
+
+Piping such a filter into another will execute the 2nd filter **_for each value_**:
+
+```sh
+$ jq -n -c '[1, 2, 3] | .[] | . * 2'
+2
+4
+6
+```
+
+This is like implicit iteration.
+Once you grok this technique, you'll realize very powerful jq filters can be very concise.
```

Comment:
```suggestion
Once you understand this technique, you'll realize very powerful jq filters can be very concise.
```

(Just because I think lots of people don't know "grok" based on my experience of confused faces when I've used it in real life!